pkg/codec/element_extractor.go

```diff
@@ -0,0 +1,91 @@
+package codec
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+type ElementExtractorLocation string
+
+const (
+	FirstElementLocation  ElementExtractorLocation = "first element"
+	MiddleElementLocation ElementExtractorLocation = "middle element"
+	LastElementLocation   ElementExtractorLocation = "last element"
```

Comment:
Is this type doing double duty as a string message describing itself? Would it make sense to use an int enum instead? with a `String()` method?
```suggestion
	FirstElementLocation  ElementExtractorLocation = iota
	MiddleElementLocation ElementExtractorLocation
	LastElementLocation   ElementExtractorLocation
```
```go
func (l ElementExtractorLocation) String() {
	...
}
```
Or just drop the common suffix?
```suggestion
	FirstElementLocation  ElementExtractorLocation = "first"
	MiddleElementLocation ElementExtractorLocation = "middle"
	LastElementLocation   ElementExtractorLocation = "last"
```

Reply:
Oh and I think that idiomatic naming here would be `ElementExtractorLocation*` instead of `*ElementLocation` - e.g. `ElementExtractorLocationFirst`, like `http.Status*` https://pkg.go.dev/net/http#pkg-constants